modules/ldap-auto-syncing.adoc

```diff
@@ -0,0 +1,246 @@
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have configured an LDAP identity provider (IDP), which involved creating an LDAP secret and config map.
+
+// TODO Any other prereqs? what about anything else w/ the sync stuff, like a config file, etc? Looks like they embed it in the config map
+
+.Procedure
+
+. Create a namespace where the cron job will run:
++
+[source,terminal]
+----
+$ oc create namespace ldap-sync <1>
+----
+<1> This procedure uses a namespace called `ldap-sync`.
+
+. Copy the secret and config map that you created when configuring the LDAP identity provider to this new namespace.
++
+The secret and config map exist in the `openshift-config` namespace and must be copied to the new `ldap-sync` namespace.
++
+// TODO: Provide actual instructions on how to copy, or just leave it like this?
+
+. Define a service account:
++
+.Example `ldap-sync-service-account.yaml`
+[source,yaml]
+----
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: ldap-group-syncer
+  namespace: ldap-sync
+  labels:
+    app: cronjob-ldap-group-sync
+----
++
+// TODO: Mention the significance of this label that is used across all objects. Is it required? Is it also needed on the config map and secret from when setting up the LDAP IDP?
+
+. Create the service account:
++
+[source,terminal]
+----
+$ oc create -f ldap-sync-service-account.yaml
+----
+
+. Define a cluster role:
++
+.Example `ldap-sync-cluster-role.yaml`
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: ldap-group-syncer
+  labels:
+    app: cronjob-ldap-group-sync
+rules:
+  - apiGroups:
+      - ''
+      - user.openshift.io
+    resources:
+      - groups
+    verbs:
+      - get
+      - list
+      - create
+      - update
+----
++
+// TODO: any of these fields need pointed out?
+
+. Create the cluster role:
++
+[source,terminal]
+----
+$ oc create -f ldap-sync-cluster-role.yaml
+----
+
+. Define a cluster role binding to bind the cluster role to the service account:
++
+.Example `ldap-sync-cluster-role-binding.yaml`
+[source,yaml]
+----
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ldap-group-syncer
+  labels:
+    app: cronjob-ldap-group-sync
+subjects:
+  - kind: ServiceAccount
+    name: ldap-group-syncer
+    namespace: ldap-sync
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: ldap-group-syncer
+----
++
+// TODO: any of these fields need pointed out?
+
+. Create the cluster role binding:
++
+[source,terminal]
+----
+$ oc create -f ldap-sync-cluster-role-binding.yaml
+----
+
+. Define a config map that specifies the LDAP config:
++
+.Example `ldap-sync-config-map.yaml`
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: ldap-group-syncer
+  namespace: ldap-sync
+  labels:
+    app: cronjob-ldap-group-sync
+data:
+  ldap-group-sync.yaml: | <1>
```

Comment:
@stlaz There are a lot of fields in this config here and I'm not sure where to start with asking questions. Can you take a look through and let me know and let me know if any might not be required, or might be worth a callout as something that need/might want to change?

Reply:
I think you got the most important ones